cmd/image-builder/README.md

```diff
@@ -202,3 +202,43 @@ This integration allows the `image-builder` binary to gather inputs directly fro
   enhanced workflow experience** comparable to the GitHub workflow approach.
 - **Build Report**: A dedicated flag (`--build-report-path`) has been added to the `image-builder` binary, enabling it to write a build
   report generated by the ADO pipeline to a specified file. This functionality is intended specifically for use in Jenkins.
+
+## Cross-Compiling and Caching for Non Native Architecture Builds
+
+Image Builder uses builder agents with `linux/amd64` native architecture.
+When building images for multiple architectures or building image for non native architecture,
+consider enabling cross-compilation to significantly reduce build times.
+Testing has shown that cross-compilation can speed up the build process by **10x**.
+Reducing build times from 12 minutes to less than 2 minutes in our tests scenario with rather small golang codebase.
+
+### Key Recommendations
+
+- **Cross-Compilation**: If you are building, builder non native architecture images, implement cross-compilation in your Dockerfile, use
+  the [Faster Multi-Platform Builds: Dockerfile Cross-Compilation Guide](https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/)
+  as a reference.
+- **Bind Mounts**: You can use bind mounts for `RUN` command in Dockerfiles to avoid copying source code for compilation.
+  Although the speed gain was minimal in our tests. We achieved speed up of less than ~5 seconds.
+- **Cache Mounts for Go Compiler**:
+  Rely on a **cache backed by a remote repository**, because a new agent is allocated for each pipeline execution, making mount-type caching
+  ineffective.
+  Use mounts a cache type for Go package downloads, and binary compilation cache did not bring speed increase during tests.
```

Comment:
```suggestion
  Use mounts a cache type for Go package downloads. The binary compilation cache did not increase speed during tests.
```

Reply:
"Use mounts a cache type" sounds very wrong to me, but maybe it's a fixed phrase. Is it? If not, please rephrase.

Reply:
This is to say to use a `mounts` that has type `cache`.